CefSharp/DevTools/DevToolsClient.cs

```diff
@@ -81,30 +81,97 @@ public void SetDevToolsObserverRegistration(IRegistration devToolsRegistration)
+                if (eventHandlers.TryGetValue(eventName, out eventRegistration))
+                {
+                    eventRegistration -= handler;
+                    if (eventRegistration == null)
+                    {
+                        eventHandlers.TryRemove(eventName, out _);
+                    }
+                }
+            });
+        }
+
+        private class Disposeable : IDisposable
+        {
+            private readonly Action _dispose;
+            public Disposeable(Action dispose)
+            {
+                _dispose = dispose;
+            }
+            public void Dispose()
+            {
+                _dispose();
+            }
+        }
+
+        /// <summary>
+        /// Execute a method call over the DevTools protocol. This method can be called on any thread.
+        /// See the DevTools protocol documentation at https://chromedevtools.github.io/devtools-protocol/ for details
+        /// of supported methods and the expected <paramref name="parameters"/> dictionary contents.
+        /// </summary>
+        /// <param name="method">is the method name</param>
+        /// <param name="parameters">are the method parameters represented as a dictionary,
+        /// which may be empty.</param>
+        /// <returns>return a Task that can be awaited to obtain the method result</returns>
+        public Task<DevToolsMethodResponse> ExecuteDevToolsMethodAsync(string method, IDictionary<string, object> parameters = null)
+        {
+            return ExecuteDevToolsMethodAsync<DevToolsMethodResponse>(method, parameters);
+        }
+
         /// <summary>
         /// Execute a method call over the DevTools protocol. This method can be called on any thread.
         /// See the DevTools protocol documentation at https://chromedevtools.github.io/devtools-protocol/ for details
         /// of supported methods and the expected <paramref name="parameters"/> dictionary contents.
         /// </summary>
+        /// <typeparam name="T">The type into which the result will be deserialzed.</typeparam>
         /// <param name="method">is the method name</param>
         /// <param name="parameters">are the method parameters represented as a dictionary,
         /// which may be empty.</param>
         /// <returns>return a Task that can be awaited to obtain the method result</returns>
-        public async Task<DevToolsMethodResponse> ExecuteDevToolsMethodAsync(string method, IDictionary<string, object> parameters = null)
+        public Task<T> ExecuteDevToolsMethodAsync<T>(string method, IDictionary<string, object> parameters = null) where T : DevToolsDomainResponseBase
         {
             if (browser == null || browser.IsDisposed)
             {
                 //TODO: Queue up commands where possible
-                return new DevToolsMethodResponse { Success = false };
+                throw new ObjectDisposedException(nameof(DevToolsClient));
```

Comment:
It's actually the browser that's been disposed in this context, the error message should reflect that.